docs/DeveloperGuide.md

```diff
@@ -7,6 +7,72 @@
 ical4J Library: https://www.ical4j.org/
 
 ## Design & implementation
+## UiRenderer Component
+### API: [UiRenderer.java](https://github.com/AY2324S2-CS2113-W13-2/tp/blob/master/src/main/java/ui/UiRenderer.java)
+
+### Overview: <br> 
+The UiRenderer component is responsible for rendering the user interface. It is used to display messages, week views as well as month views to the user.
+
+### How it works:
+1. When week view is requested, the UiRenderer component will render the week view.
+2. This is done by printing the table in ASCII art format, with the days of the week as columns.
+3. The week view will display the tasks for the week, with the tasks for each day displayed in the respective columns.
+4. The UiRenderer component will also display the current date at the top of the week view.
+5. Similarly, when month view is requested, the UiRenderer component will render the month view.
+
+### printWeekHeader Method
+`printWeekHeader` Method
+
+The `printWeekHeader` method is responsible for rendering the header section of the week view, including the names of the days and optionally the dates.
+
+#### Method signature:
+```
+public static void printWeekHeader(LocalDate startOfView, DateTimeFormatter dateFormatter, boolean isMonthView)
+```
+
+#### Parameters
+- startOfView: The starting date of the week view.
+- dateFormatter: Formatter for displaying dates.
+- isMonthView: A boolean flag indicating if the month view is being rendered.
```

Comment:
As mentioned by Clarence on my PR, we should discuss whether we should include details like the parameters of the method